### DISCUSSION TITLE
[INFO] No SMBus device found

### DISCUSSION THREAD

**Discussion by hepike** [laszlodaniel/SmartBatteryHack#17] [General]
Hi,

I have some trouble connecting to a battery. I have 3 different type of dell battery: 
- T54FJ (connected once on address 0B, manufacturer and device name were correct)
- NHXVW (connected once, but found 7 addresses [03-09]. Tried everyone but only got zeros)
- N3X1D (connected once on address 0B, read the manufacturer correctly, then this manufacturer data appeared on every register when I tried to read)

And thats it, now I can not connect to any of my batteries. Checked the cables and ressistors (1k for battery pin 1 and 4, and 4k7 on both SDA and SCL lines to the 5V). Tried to reprogram the Arduino Uno R3 with SmartBatteryHack_SoftI2CMaster.ino and SmartBatteryHack_Wire.ino as well, but no luck. Btw, which one should I upload to the arduino?

Can you help me?

Thank you,
Adam.

**Comment by laszlodaniel** [comment]
Hi, I have the same issue recently. In the past, everything was fine, then suddenly this behavior appeared.
You should use SmartBatteryHack_SoftI2CMaster.ino. It uses the library meant to be used with SMBus.
This issue doesn't seem to be intermittent: it works once and then never again, like it was a hardware problem.
I don't know what to think about this.

**Reply by jaro44** [reply]
Correct everything, after the latest changes i am also randomly connecting to my battery.
Moreover, after a few seconds how to connect disconnects itself and there is no smbus connection

**Reply by jaro44** [reply]
edit:
however, I think I broke my battery electronics,
because on SMBus interface based on the Cypress FX2LP with SMBUSB I have the same :(

**Reply by laszlodaniel** [reply]
Do you have an idea what might caused the brick?
I also destroyed my original test battery by connecting 12.6V to its terminals. I don't know how that happened but magic smoke was released. So I was expecting it not to work with this reader.
Then I tried it on 2 other untouched batteries with no luck. Sometimes another I2C scanner discovers the battery address and then nothing.
I can't even decide if this is software or hardware related issue.

**Reply by jaro44** [reply]
in my battery the A2168 chip seems ok,
I need to review the rest of the electronics, diodes, resistors, capacitors

**Reply by hepike** [reply]
Oh, okay. I suspected user or hw error, this is why i didn't open issue. I am going to test the new release and let you know.

**Comment by laszlodaniel** [comment]
I identified a pinout problem with ATmega2560 (SDA and SCL were reversed... big blunder) and added the I2C_NOINTERRUPT flag to the sketch. Just tested with Arduino Uno and Arduino Mega and it works! Finally.
I'm uploading the modified sketch now.

**Reply by jaro44** [reply]
What was the problem to replace the wires.
Unfortunately I have a hardware battery problem.

**Reply by laszlodaniel** [reply]
I'm sorry to hear that.
The wires were okay, the sketch had the PORTD pins the other way around. I must have switched them sometime recently. Now all is fixed.

**Reply by jaro44** [reply]
I only remained with blocked BQ8050 battery, unfortunately, I don't know how to unseal.
I have permanently 0% and does not charging :(

**Comment by lmdpua** [comment]
I made several devices for reading the characteristics of batteries on an arduino.  Now I'm doing the same project, but on stm32.  I have noticed that some batteries are poorly readable.  First, this is due to the pull-up resistors.  They should be in the range of 2-4.7 kOhm.
Secondly.  Some batteries read well at 200kHz, while Lenovo is readable only at 50kHz and below.  Through experiments, I lowered the frequency to 20 kHz.  Fortunately, the development environment allows you to do this in a few clicks.  All batteries can now be read confidently.
I also recommend making an optical isolation.  By negligence, I have already burned several arduino boards and cp2112 boards
